src/main/java/de/tum/in/www1/artemis/service/connectors/iris/IrisConnectorService.java

```diff
@@ -111,11 +111,11 @@ private <Response> CompletableFuture<Response> sendRequestAndParseResponse(Strin
     }
 
     private IrisException toIrisException(HttpStatusCodeException e, String preferredModel) {
-        return switch (e.getStatusCode()) {
-            case UNAUTHORIZED, FORBIDDEN -> new IrisForbiddenException();
-            case BAD_REQUEST -> new IrisInvalidTemplateException(tryExtractErrorMessage(e));
-            case NOT_FOUND -> new IrisModelNotAvailableException(preferredModel, tryExtractErrorMessage(e));
-            case INTERNAL_SERVER_ERROR -> new IrisInternalPyrisErrorException(tryExtractErrorMessage(e));
+        return switch (e.getStatusCode().value()) {
+            case 401, 403 -> new IrisForbiddenException();
```

Comment:
why do you change it from contants to numbers?

Reply:
Because it annoyingly got changed in Spring Boot 3. I have no idea why they decided to stick with ints instead of Enums.